contributors/devel/development.md

```diff
@@ -101,35 +119,56 @@ revising your hypothesis.  The above issues do this transparently, using figures
 and data rather then conjecture. Notice that the problem is analyzed and a
 correct solution is created before a single line of code is reviewed.
 
-## Building Kubernetes with Docker
+## Getting the code
+
+Kubernetes is developed using git for version control.
+
+### GitHub workflow
+
+To check out code to work on, please refer to [this guide](/contributors/guide/github-workflow.md).
+
+For tips, tricks, and common practices used within the Kubernetes project read the [Kubernetes GitHub Cheat Sheet][github-cheat-sheet].
+
+## Building Kubernetes
+
+There are two recommended ways of building Kubernetes, and the choice of one or the other depends on if you do plan on developing upstream. One way is building a release from source, this is recommended for development purposes. If there is no development involved, it is suggested to use a pre-built version of the current release.
+
+### Building Kubernetes with Docker
 
 Official releases are built using Docker containers. To build Kubernetes using
 Docker please follow [these
 instructions](http://releases.k8s.io/HEAD/build/README.md).
```

Comment:
See comment below: building with Docker and locally... _seem_ like it's the same thing?

Reply:
also you may want to link to here; the above link doesn't link to `master`: http://git.k8s.io/kubernetes/build/README.md

Reply:
Link updated.

Reply:
The locally part is when you aren't using Docker. This one is the recommended if you are going to develop or contribute upstream.